### PR TITLE
Fix build with `#[unix_sigpipe = "..."]` support in rustc

### PR DESCRIPTION
**Pull request by Enselic**
Closes https://github.com/rust-lang/rust/issues/101352

CC @RalfJung @oli-obk

### CODE REVIEW

**Comment by RalfJung** [comment]
Could you update the rust-version file? Right now it is still testing the old rustc.

I think further adjustment will be needed, since Miri needs to pass the right number of arguments to the start fn.

**Comment by Enselic** [comment]
Yeah I was a bit rushed before and only tried to build and not run tests. Running tests revealed at least one more problem.

This is not a complete solution though because tests are still failing locally for me, but it takes one more step in the right direction.

File: src/eval.rs [hunk 1]
```diff
@@ -301,10 +301,12 @@ pub fn create_ecx<'mir, 'tcx: 'mir>(
             )?;
         }
         EntryFnType::Start => {
+            let sigpipe = 2; // Inlining of `DEFAULT` from https://github.com/rust-lang/rust/blob/master/compiler/rustc_session/src/config/sigpipe.rs
+            let sigpipe_arg = Scalar::from_u8(sigpipe);
             ecx.call_function(
```
Comment (RalfJung) [inline src/eval.rs:306]: There's another `call_function` a few lines up, that one probably also needs to be adjusted.

Comment (Enselic) [reply src/eval.rs]: Oops, yes, I got it backwards in my head, I should change that place *instead* of this place.

All tests pass locally for me now, hopefully it all works in CI too.

**Comment by RalfJung** [comment]
Thanks!
@bors r+

**Pull request merged**